WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipment Tracking Section/Add Tracking/ManualTrackingViewController.swift

```diff
@@ -134,7 +136,13 @@ private extension ManualTrackingViewController {
     }
 
     @objc func dismissButtonTapped() {
-        dismiss()
+        if viewModel.canCommit || viewModel.hasUnsavedChanges {
```

Comment:
Do we need to check both `viewModel.canCommit || viewModel.hasUnsavedChanges` here, or `viewModel.hasUnsavedChanges` would be enough?

Reply:
At the moment we need to check both to cover the case of "User do not fill any field, but taps dismiss", this would trigger the AlertSheet which we do not want, because there's actually nothing to discard there, can be dismissed directly. This is something that should be fixed on `hasUnsavedChanges` logic though, as you said, if both fields are empty this would return true, which isn't correct.